packages/compass-e2e-tests/index.js

```diff
@@ -35,11 +39,36 @@ async function main() {
     mocha.addFile(path.join(__dirname, testPath));
   });
 
-  mocha.run((failures) => (process.exitCode = failures ? 1 : 0));
+  keychain.activate();
+
+  mocha.run((failures) => {
+    keychain.reset();
+    process.exitCode = failures ? 1 : 0;
+  });
 }
 
+process.on('SIGINT', () => {
+  debug(`Process was interrupted. Resetting keychain to default and exiting.`);
+  keychain.reset();
+  process.exitCode = 128 + SIGINT;
+  process.exit();
+});
+
+process.on('SIGTERM', () => {
+  debug(`Process was terminated. Resetting keychain to default and exiting.`);
+  keychain.reset();
+  process.exitCode = 128 + SIGTERM;
+  process.exit();
+});
+
+process.on('uncaughtException', () => {
+  debug('Uncaught exception. Resetting keychain to default and exiting.');
```

Comment:
I guess the exiting in this case happens elsewhere?

Reply:
Oh, I was convinced for some reason that adding an event listener to `uncaughtException` doesn't override default Node.js behavior and the process will exit as soon as custom handler is evaluated. [Double-checked with the docs](https://nodejs.org/api/process.html#process_event_uncaughtexception) and it's not the case

Reply:
Done in 2dfe051ca6d8eb73870530623d9a8e797456344b